lib/src/react_client/event_helpers.dart

```diff
@@ -0,0 +1,324 @@
+    properties['currentTarget'] ?? baseEvent?.currentTarget,
+    properties['defaultPrevented'] ?? baseEvent?.defaultPrevented,
+    properties['preventDefault'] ?? baseEvent?.preventDefault ?? () {},
+    properties['stopPropagation'] ?? baseEvent?.stopPropagation ?? () {},
+    properties['eventPhase'] ?? baseEvent?.eventPhase,
+    properties['isTrusted'] ?? baseEvent?.isTrusted,
+    properties['nativeEvent'] ?? baseEvent?.nativeEvent,
+    properties['target'] ?? baseEvent?.target,
+    properties['timeStamp'] ?? baseEvent?.timeStamp ?? 0,
+    properties['type'] ?? baseEvent?.type ?? 'empty event',
+    properties['animationName'] ?? baseEvent?.animationName,
+    properties['elapsedTime'] ?? baseEvent?.elapsedTime,
+    properties['pseudoElement'] ?? baseEvent?.pseudoElement,
+  );
+}
+
+SyntheticUIEvent buildUIEvent({SyntheticUIEvent baseEvent, Map<String, dynamic> properties = const {}}) {
+  return SyntheticUIEvent(
+    properties['bubbles'] ?? baseEvent?.bubbles,
+    properties['cancelable'] ?? baseEvent?.cancelable,
+    properties['currentTarget'] ?? baseEvent?.currentTarget,
+    properties['defaultPrevented'] ?? baseEvent?.defaultPrevented,
+    properties['preventDefault'] ?? baseEvent?.preventDefault ?? () {},
+    properties['stopPropagation'] ?? baseEvent?.stopPropagation ?? () {},
+    properties['eventPhase'] ?? baseEvent?.eventPhase,
+    properties['isTrusted'] ?? baseEvent?.isTrusted,
+    properties['nativeEvent'] ?? baseEvent?.nativeEvent,
+    properties['target'] ?? baseEvent?.target,
+    properties['timeStamp'] ?? baseEvent?.timeStamp ?? 0,
+    properties['type'] ?? baseEvent?.type ?? 'empty event',
+    properties['detail'] ?? baseEvent?.detail,
+    properties['view'] ?? baseEvent?.view,
+  );
+}
+
+SyntheticWheelEvent buildWheelEvent({SyntheticWheelEvent baseEvent, Map<String, dynamic> properties = const {}}) {
+  return SyntheticWheelEvent(
+    properties['bubbles'] ?? baseEvent?.bubbles,
+    properties['cancelable'] ?? baseEvent?.cancelable,
+    properties['currentTarget'] ?? baseEvent?.currentTarget,
+    properties['defaultPrevented'] ?? baseEvent?.defaultPrevented,
+    properties['preventDefault'] ?? baseEvent?.preventDefault ?? () {},
+    properties['stopPropagation'] ?? baseEvent?.stopPropagation ?? () {},
+    properties['eventPhase'] ?? baseEvent?.eventPhase,
+    properties['isTrusted'] ?? baseEvent?.isTrusted,
+    properties['nativeEvent'] ?? baseEvent?.nativeEvent,
+    properties['target'] ?? baseEvent?.target,
+    properties['timeStamp'] ?? baseEvent?.timeStamp ?? 0,
+    properties['type'] ?? baseEvent?.type ?? 'empty event',
+    properties['deltaX'] ?? baseEvent?.deltaX,
+    properties['deltaMode'] ?? baseEvent?.deltaMode,
+    properties['deltaY'] ?? baseEvent?.deltaY,
+    properties['deltaZ'] ?? baseEvent?.deltaZ,
+  );
+}
+
+extension SyntheticEventTypeHelpers on SyntheticEvent {
+  bool get isClipboardEvent => this.tryCast<SyntheticClipboardEvent>() != null;
```

Comment:
#nit tryCast isn't necessary here
```suggestion
  bool get isClipboardEvent => this is SyntheticClipboardEvent;
```

Reply:
[But proxies, Greg, proxy all the things!](https://github.com/EnterpriseQualityCoding/FizzBuzzEnterpriseEdition/blob/uinverse/src/main/java/com/seriouscompany/business/java/fizzbuzz/packagenamingpackage/impl/Main.java)

Jk 🤦